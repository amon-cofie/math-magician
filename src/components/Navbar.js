import React from 'react';
import { NavLink } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';

const Navbar = () => {
  const links = [
    {
      id: uuidv4(),
      path: '/',
      text: 'Home',
    },
    {
      id: uuidv4(),
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: uuidv4(),
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <nav className="navbar-container">
      <h1 className="main-heading">Math Magicians</h1>
      <ul className="nav-links">
        {links.map((link) => (
          <li data-testid={link.text} key={link.id}>
            <NavLink to={link.path}>{link.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
