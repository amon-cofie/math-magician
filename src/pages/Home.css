.home-container {
  font-size: 1.5rem;
  display: flex;
  flex-direction: column;
  min-height: 70vh;
  padding: 2rem;
}
