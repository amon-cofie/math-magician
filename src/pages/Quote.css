.quote-container {
  font-size: 2rem;
  display: flex;
  text-align: center;
  align-items: center;
  min-height: 70vh;
  padding: 2rem;
}
