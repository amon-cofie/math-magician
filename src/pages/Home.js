import React from 'react';
import './Home.css';

function Home() {
  return (
    <div className="home-container">
      <h1>Welcome to our page!</h1>
      <br />
      <br />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos
        porro quis nemo dolore autem unde voluptatum nostrum vel nobis culpa
        aliquam totam temporibus atque accusantium blanditiis sunt officiis
        sequi, libero adipisci voluptates et eum incidunt quos corrupti. Modi
        optio illo nulla aut at tempore corporis unde tempora, vero quasi vel.
      </p>
      <br />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos
        porro quis nemo dolore autem unde voluptatum nostrum vel nobis culpa
        aliquam totam temporibus atque accusantium blanditiis sunt officiis
        sequi, libero adipisci voluptates et eum incidunt quos corrupti. Modi
        optio illo nulla aut at tempore corporis unde tempora, vero quasi vel.
      </p>
    </div>
  );
}

export default Home;
