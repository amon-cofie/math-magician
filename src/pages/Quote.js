import React from 'react';
import './Quote.css';

function Quote() {
  return (
    <div className="quote-container">
      <p>
        Mathematics is not all about numbers, equations, computations, or
        algorithms: it is about understanding. -William Paul Thurston
      </p>
    </div>
  );
}

export default Quote;
