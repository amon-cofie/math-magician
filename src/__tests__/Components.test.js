import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
import Home from '../pages/Home';
import Quote from '../pages/Quote';

test('renders navbar component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders Home component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders About component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Quote />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
