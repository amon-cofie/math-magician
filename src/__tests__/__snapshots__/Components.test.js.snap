// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders About component snapshot 1`] = `
<DocumentFragment>
  <div
    class="quote-container"
  >
    <p>
      Mathematics is not all about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
    </p>
  </div>
</DocumentFragment>
`;

exports[`renders Home component snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <h1>
      Welcome to our page!
    </h1>
    <br />
    <br />
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos porro quis nemo dolore autem unde voluptatum nostrum vel nobis culpa aliquam totam temporibus atque accusantium blanditiis sunt officiis sequi, libero adipisci voluptates et eum incidunt quos corrupti. Modi optio illo nulla aut at tempore corporis unde tempora, vero quasi vel.
    </p>
    <br />
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos porro quis nemo dolore autem unde voluptatum nostrum vel nobis culpa aliquam totam temporibus atque accusantium blanditiis sunt officiis sequi, libero adipisci voluptates et eum incidunt quos corrupti. Modi optio illo nulla aut at tempore corporis unde tempora, vero quasi vel.
    </p>
  </div>
</DocumentFragment>
`;

exports[`renders navbar component snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar-container"
  >
    <h1
      class="main-heading"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-links"
    >
      <li
        data-testid="Home"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        data-testid="Calculator"
      >
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        data-testid="Quote"
      >
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
